SajidHossain-submission-improved/tests/test_recursive_binary_search.py

```diff
@@ -0,0 +1,25 @@
+
+import unittest
```

Comment:
These unit test files did not run successfully when I followed the instructions that you provided! 

I had to insert the following lines to be able to run them.

```suggestion
import unittest

import os
import sys
sys.path.append(os.getcwd() + '/..')
```
And they failed the tests :(
```
EEEEEE
======================================================================
ERROR: test_element_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 20, in test_element_found
    self.assertEqual(recursive_binary_search([1, 2, 3, 4, 5], 3), 2)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_element_not_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 23, in test_element_not_found
    self.assertEqual(recursive_binary_search([1, 2, 3, 4, 5], 6), -1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_empty_list (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 11, in test_empty_list
    self.assertEqual(recursive_binary_search([], 5), -1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_negative_numbers (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 26, in test_negative_numbers
    self.assertEqual(recursive_binary_search([-5, -3, 0, 2, 4], -3), 1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_single_element_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 14, in test_single_element_found
    self.assertEqual(recursive_binary_search([5], 5), 0)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_single_element_not_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 17, in test_single_element_not_found
    self.assertEqual(recursive_binary_search([5], 3), -1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

----------------------------------------------------------------------
Ran 6 tests in 0.002s

FAILED (errors=6)
(base) daniel@DansComputer:~/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests$ python test_recursive_binary_search.py >output.txt
EEEEEE
======================================================================
ERROR: test_element_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 20, in test_element_found
    self.assertEqual(recursive_binary_search([1, 2, 3, 4, 5], 3), 2)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_element_not_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 23, in test_element_not_found
    self.assertEqual(recursive_binary_search([1, 2, 3, 4, 5], 6), -1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_empty_list (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 11, in test_empty_list
    self.assertEqual(recursive_binary_search([], 5), -1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_negative_numbers (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 26, in test_negative_numbers
    self.assertEqual(recursive_binary_search([-5, -3, 0, 2, 4], -3), 1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_single_element_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 14, in test_single_element_found
    self.assertEqual(recursive_binary_search([5], 5), 0)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

======================================================================
ERROR: test_single_element_not_found (__main__.TestRecursiveBinarySearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/daniel/Documents/MAT2440-Projects-F24/SajidHossain-submission-improved/tests/test_recursive_binary_search.py", line 17, in test_single_element_not_found
    self.assertEqual(recursive_binary_search([5], 3), -1)
TypeError: recursive_binary_search() missing 2 required positional arguments: 'low' and 'high'

----------------------------------------------------------------------
Ran 6 tests in 0.001s

FAILED (errors=6)
```